realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java

```diff
@@ -167,7 +167,7 @@ public boolean hasPrimaryKey(Class<? extends RealmModel> clazz){
      * @param object {@link RealmObject} to insert.
      * @param cache the cache for mapping between unmanaged objects and their table row index for eventual reuse.
      */
-    public abstract void insert(Realm realm, RealmModel object, Map<RealmModel, Long> cache);
+    public abstract Long insert(Realm realm, RealmModel object, Map<RealmModel, Long> cache);
```

Comment:
Can this be null? Otherwise use primitive long.

Reply:
I don't know why but my eyes read the insert in the proxies as `Long`.